boa/src/syntax/ast/node/declaration/async_function_decl/mod.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{
+    exec::Executable,
+    syntax::ast::node::{join_nodes, FormalParameter, Node, StatementList},
+    BoaProfiler, Context, Result, Value,
+};
+use gc::{Finalize, Trace};
+use std::fmt;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+/// Async Function Declaration.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///  - [MDN documentation][mdn]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-async-function-prototype-properties
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Clone, Debug, Trace, Finalize, PartialEq)]
+pub struct AsyncFunctionDecl {
+    name: Option<Box<str>>,
+    parameters: Box<[FormalParameter]>,
+    body: StatementList,
+}
+
+impl AsyncFunctionDecl {
+    /// Creates a new async function declaration.
+    pub(in crate::syntax) fn new<N, P, B>(name: N, parameters: P, body: B) -> Self
+    where
+        N: Into<Option<Box<str>>>,
+        P: Into<Box<[FormalParameter]>>,
+        B: Into<StatementList>,
+    {
+        Self {
+            name: name.into(),
+            parameters: parameters.into(),
+            body: body.into(),
+        }
+    }
+
+    /// Gets the name of the async function declaration.
+    pub fn name(&self) -> Option<&str> {
+        self.name.as_deref()
+    }
+
+    /// Gets the list of parameters of the async function declaration.
+    pub fn parameters(&self) -> &[FormalParameter] {
+        &self.parameters
+    }
+
+    /// Gets the body of the async function declaration.
+    pub fn body(&self) -> &[Node] {
+        self.body.statements()
+    }
+
+    /// Implements the display formatting with indentation.
+    pub(in crate::syntax::ast::node) fn display(
+        &self,
+        f: &mut fmt::Formatter<'_>,
+        indentation: usize,
+    ) -> fmt::Result {
+        match &self.name {
+            Some(name) => {
+                write!(f, "async function {}(", name)?;
+            }
+            None => {
+                write!(f, "async function (")?;
+            }
+        }
+        join_nodes(f, &self.parameters)?;
+        f.write_str(") {{")?;
+
+        self.body.display(f, indentation + 1)?;
+
+        writeln!(f, "}}")
+    }
+}
+
+impl Executable for AsyncFunctionDecl {
+    fn run(&self, _: &mut Context) -> Result<Value> {
+        let _timer = BoaProfiler::global().start_event("AsyncFunctionDecl", "exec");
+        // unimplemented!("Execute AsyncFunctionDecl");
```

Comment:
```suggestion
        // TODO: Implement AsyncFunctionDecl
```
Better replace this with a todo